File: tutor-claude/msa-tutor/auth-service/src/main/java/be/com/msatutor/auth/api/AuthController.java
package be.com.msatutor.auth.api;

import be.com.msatutor.auth.domain.AuthService;
import jakarta.validation.Valid;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/auth")
public class AuthController {

    private final AuthService authService;

    public AuthController(AuthService authService) {
        this.authService = authService;
    }

    @PostMapping("/register")
    public AuthDto.TokenResponse register(@Valid @RequestBody AuthDto.RegisterRequest request) {
        String token = authService.register(request);
        return new AuthDto.TokenResponse(token, "Bearer");
    }

    @PostMapping("/login")
    public AuthDto.TokenResponse login(@Valid @RequestBody AuthDto.LoginRequest request) {
        String token = authService.login(request);
        return new AuthDto.TokenResponse(token, "Bearer");
    }
}

File: tutor-claude/spring-security-tutor/src/main/kotlin/be/com/springsecuritytutor/auth/AuthController.kt
package be.com.springsecuritytutor.auth

import be.com.springsecuritytutor.security.UserPrincipal
import be.com.springsecuritytutor.user.UserResponse
import jakarta.validation.Valid
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestHeader
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/auth")
class AuthController(
    private val authService: AuthService
) {

    @PostMapping("/register")
    fun register(@Valid @RequestBody request: RegisterRequest): TokenResponse {
        return authService.register(request)
    }

    @PostMapping("/login")
    fun login(@Valid @RequestBody request: LoginRequest): TokenResponse {
        return authService.login(request)
    }

    @PostMapping("/refresh")
    fun refresh(@Valid @RequestBody request: RefreshRequest): TokenResponse {
        return authService.refresh(request)
    }

    @PostMapping("/logout")
    fun logout(
        @Valid @RequestBody request: LogoutRequest,
        @RequestHeader("Authorization", required = false) authorization: String?
    ) {
        val accessToken = authorization?.takeIf { it.startsWith("Bearer ") }?.removePrefix("Bearer ")?.trim()
        authService.logout(request.refreshToken, accessToken)
    }

    @GetMapping("/me")
    fun me(@AuthenticationPrincipal principal: UserPrincipal): UserResponse {
        return UserResponse.from(principal.user)
    }
}
